testing/functional/tests/cypress/integration/student/01-welcome-page.spec.ts

```diff
@@ -0,0 +1,47 @@
+import WelcomeObject from "../../page-objects/student-objects/WelcomeObject"
+
+describe("Welcome Page", () => {
+
+    const welcome = new WelcomeObject()
+    beforeEach(function () {
+        cy.visit("/")
+    })
+
+    it("Validate student URL",()=>{
+        welcome.checkCurrentUrl()
+    })
+
+    it("Validate welcome page", () => {
+        welcome.getWelcomeText()
+    })
+
+    it("Validate Login with BCSC & Sign Up with BCSC buttons are visible",()=>{
+        welcome.loginWithBCSCBtn()
+        welcome.signUpWithBCSCBtn()
+    })
+
+    it("Verify that clicking on Login with BCSC button redirects to appropriate page or not",()=>{
+        welcome.verifyLoginBtn()
+    })
+
+    it("Verify that clicking on Sign Up with BCSC button redirects to appropriate page or not",()=>{
+        welcome.verifySignUpBtn()
```

Comment:
I would suggest that, in general, the page objects would be used to map the page objects, while the assertions would be executed on the specific tests in this file. Does it make sense?
Is there a pattern being followed here that would show the benefits of doing it in this way?

Reply:
I have created page objects for all specs files. The purpose of creating page objects is to define multiple selectors in single function. Since multiple selectors are being utilized, we pass only the functions in the spec files. This means in the future, when updating selectors, we must change from the single-function approach.

Attached page object scenario:- 
I have created function for application button in Dashboard Page object.
<img width="620" alt="image" src="https://user-images.githubusercontent.com/98920571/158674690-dcbe6e95-06b3-4d7c-b37d-5c86a3de4ee1.png">

I am using that application button in Dashboard as well as in Application specs file. I don't need to write two times here, I have to just pass the function.
<img width="685" alt="image" src="https://user-images.githubusercontent.com/98920571/158675417-e5d441c0-63ce-481f-83e5-e2dca0005985.png">

Reply:
So, I believe that we are saying the same about this: "I have created page objects for all specs files. The purpose of creating page objects is to define multiple selectors in single function."

My point is that the Page objects are doing what you mentioned above **and also doing the validation (assertions)**.
![image](https://user-images.githubusercontent.com/61259237/158703902-c236edc3-fed7-4224-80cc-f2a38a9118bf.png)

My point was:
1. Use the page objects to "define multiple selectors", as you mentioned
2. Do the checks in the method calling the page object, the one actually doing the test.

Reply:
Thanks for doing these changes 😉